Userland/Applications/DisplaySettings/BackgroundSettingsWidget.cpp

```diff
@@ -50,20 +50,93 @@ BackgroundSettingsWidget::BackgroundSettingsWidget(bool& background_settings_cha
 {
 }
 
+/// A thin wrapper over FileSystemModel for wallpapers in /res/wallpapers. The
+/// only change from a plain FileSystemModel is row zero reserved for "None"
+/// (and this assumes no nested directories).
+class WallpapersModel final : public GUI::Model
```

Comment:
Seems a little awkward to make a model just for this, but `FileSystemModel` handles generating the thumbnails, which was easier to reuse.